diff/merge_allof.go

```diff
@@ -0,0 +1,868 @@
+/*package openapi3*/
+package diff
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"math"
+	"regexp"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+const (
+	FormatErrorMessage = "unable to resolve Format conflict: all Format values must be identical"
+	TypeErrorMessage   = "unable to resolve Type conflict: all Type values must be identical"
+
+	formatInt32  = "int32"
+	formatInt64  = "int64"
+	formatFloat  = "float"
+	formatDouble = "double"
+)
+
+type SchemaCollection struct {
+	Not                  []*openapi3.SchemaRef
+	OneOf                []openapi3.SchemaRefs
+	AnyOf                []openapi3.SchemaRefs
+	Title                []string
+	Type                 []string
+	Format               []string
+	Description          []string
+	Enum                 [][]interface{}
+	UniqueItems          []bool
+	ExclusiveMin         []bool
+	ExclusiveMax         []bool
+	Min                  []*float64
+	Max                  []*float64
+	MultipleOf           []*float64
+	MinLength            []uint64
+	MaxLength            []*uint64
+	Pattern              []string
+	MinItems             []uint64
+	MaxItems             []*uint64
+	Items                []*openapi3.SchemaRef
+	Required             [][]string
+	Properties           []openapi3.Schemas
+	MinProps             []uint64
+	MaxProps             []*uint64
+	AdditionalProperties []openapi3.AdditionalProperties
+}
+
+// Merge replaces objects under AllOf with a flattened equivalent
+func Merge(schema openapi3.Schema) (*openapi3.Schema, error) {
```

Comment:
Can we rename this to 'merge' so it's a private function in diff package?

Reply:
We can't do that because it's used in diff_test package